src/index.test.ts

```diff
@@ -334,6 +335,131 @@ describe('batchQueue', () => {
   })
 })
 
+describe('batchQueueParallel', () => {
+  test('should batch items up to batchSize', () => {
+    expect.assertions(3)
+    const calls: any[] = []
+    const fn = (records: string[]) => {
+      calls.push([...records]) // Copy array since it gets reset
+    }
+    const batchSize = 2
+
+    const queue = batchQueueParallel(fn, { batchSize })
+    const records = ['Joe', 'Frank', 'Bob']
+
+    for (const record of records) {
+      queue.enqueue(record)
+    }
```

Comment:
It might be worth testing what happens when the `queue.enqueue` calls happen in parallel, e.g.:

```typescript
    for (const record of records) {
      (async () => queue.enqueue(record))()
    }
```

My thought is that the tests could simulate what happens when you combine `batchQueueParallel` and `rateLimit`.

Or maybe we could have a test case that actually uses those two things together?

Reply:
Yeah, I just saw that! Nice.

Reply:
You can add `.only` to that test and run `DEBUG=prom-utils:* npm test` to see the order of things.